### DISCUSSION TITLE
Route Groups do not work within [locale]

### DISCUSSION THREAD

**Discussion by BearCooder** [amannn/next-intl#800] [Q&A]
Hello I have a strange issue. It seems that [Route Groups](https://nextjs.org/docs/app/building-your-application/routing/route-groups) do not work with next-intl?

My following structure:
```
app/
 [locale]/
  (dashboard)/
    customers/
      page.tsx
    summary/
      page.tsx
   layout.tsx

```

Now when I want to access the customers page on `http://localhost:3000/customers` or `http://localhost:3000/summary` I get a 404 not found error.  I have to use `http://localhost:3000/dashboard/customers` to access the page.  But when I try it without the [locale] then it works as expected and I can access `http://localhost:3000/customers` without dashboard. When I put it all back into [locale] folder it does not work again?

I am not sure what causes this issue? I looked at the documentation in next intl and couldnt find anything related to this.

**Comment by amannn** [comment]
I haven't worked with route groups so far, but if you can reproduce the bug in [a Next.js-only app](https://codesandbox.io/p/devbox/next-intl-nextjs-repro-3z3pjk?file=%2Fsrc%2Fapp%2F%5Blocale%5D%2Fpage.tsx) then you might want to submit this as a bug report in the Next.js repo.

**Comment by J4v4Scr1pt** [comment]
Same issue here when multiple root layouts.

**Comment by astronaute77** [comment]
I have the same problem, I think the issue is in `setRequestLocale(locale);` executed in the layout. If it is executed from a route group, the `let locale = await requestLocale;` in request.ts returns "undefined"

**Comment by pbortnick** [comment]
Strongly suggest including a route group example in the docs 🙏

**Comment by amannn** [comment]
As a side note here, the Next.js team is working on [`rootParams`](https://github.com/vercel/next.js/pull/72837), an API that `next-intl` will benefit of greatly, allowing to get rid of `await requestLocale` in favor of a built-in feature in Next.js. It's also expected to work with route groups, with multiple root layouts.

I hope to have more to share on this soon, once the API is available on the stable channel of Next.js!

**Comment by Yanick64** [comment]
Is there any update on the route group part? 
Someone else went for a different approach?

Types seems incorrectly typed/missing groups within the `defineRouting`.
